compiler/vmmarshal.nim

```diff
@@ -36,8 +36,8 @@ proc getField(n: PNode; position: int): PSym =
 proc storeAny(s: var string; t: PType; a: PNode; stored: var IntSet)
 
 proc storeObj(s: var string; typ: PType; x: PNode; stored: var IntSet) =
```

Comment:
storeObj is called only on nodes with tyObject types. With this PR it is now conceptually wrong to treat objects as nkPar (tuple) nodes.